src/NuGet.Clients/NuGet.VisualStudio.Common/IDE/VsHierarchyUtility.cs

```diff
@@ -155,6 +173,47 @@ public static async Task CollapseAllNodesAsync(IDictionary<string, ISet<VsHierar
             }
         }
 
+        public static List<IVsHierarchy> GetAllLoadedProjects(IVsSolution vsSolution)
+        {
+            Assumes.NotNull(vsSolution);
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            if (!IsSolutionOpen(vsSolution))
+            {
+                return new List<IVsHierarchy>();
```

Comment:
Honestly, this is more of a theoretical perf increase, rather than something that will have measurable, let alone meaningful, increase.  But, if you change the method's return type to `IReadOnlyList<IVsHierarchy>`, then I think this like can be `Array.Empty()`, and save an allocation.

Reply:
Actually, given that there's only usage of the method, we're checking the solution open twice and there's simply no need for that.